pandas/tests/extension/base/dim2.py

```diff
@@ -20,6 +20,17 @@ class Dim2CompatTests:
     # Note: these are ONLY for ExtensionArray subclasses that support 2D arrays.
     #  i.e. not for pyarrow-backed EAs.
 
+    @pytest.fixture(autouse=True)
+    def skip_if_doesnt_support_2d(self, dtype, request):
+        if not dtype._supports_2d:
+            node = request.node
+            # In cases where we are mixed in to ExtensionTests, we only want to
+            #  skip tests that are defined in Dim2CompatTests
+            test_func = node._obj
+            if test_func.__qualname__.startswith("Dim2CompatTests"):
+                # TODO: is there a less hacky way of checking this?
```

Comment:
Could the tests in `Dim2CompatTests` just be skipped if `data.dtype._supports_2d`?

Reply:
this is skipping if _not_ `data.dtype.supports_2d`.  The acrobatics here are for test classes that subclass ExtensionTests, since we want this fixture to only apply to tests defined in Dim2CompatTests, not all tests in ExtensionTests